From 34841b5bd8ed7979bd7828829b790d511fe08733 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 8 Dec 2017 14:54:26 +0800
Subject: [PATCH 3015/5242] MLK-17126-1: hdp api: workaround for imx8qm HDMI
 DDC R/W issue

commit  2fc31728be1a65404f4162bcf194607d74f9a55d from
https://source.codeaurora.org/external/imx/linux-imx.git

HDMI DDC R/W function is not supported by imx8qm HDMI FW.
Skip the function for imx8qm before the issue is fixed in FW.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_HDMITX.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp/API_HDMITX.c b/drivers/mxc/hdp/API_HDMITX.c
index b1f3a78..9d1f63d 100644
--- a/drivers/mxc/hdp/API_HDMITX.c
+++ b/drivers/mxc/hdp/API_HDMITX.c
@@ -52,6 +52,7 @@
 #include "address.h"
 #include "source_car.h"
 #include "source_vif.h"
+#include <soc/imx8/soc.h>
 
 CDN_API_STATUS CDN_API_HDMITX_DDC_READ(state_struct *state,
 				       HDMITX_TRANS_DATA *data_in,
@@ -159,7 +160,11 @@ CDN_API_STATUS CDN_API_HDMITX_READ_EDID_blocking(state_struct *state, u8 block,
 	data_in.len = 1;
 	data_in.slave = 0x54;
 	data_in.offset = 0x20;	/* TMDS config */
-	ret = CDN_API_HDMITX_DDC_WRITE_blocking(state, &data_in, &data_out);
+	/* Workaround for imx8qm DDC R/W failed issue */
+	if (!cpu_is_imx8qm()) {
+		ret = CDN_API_HDMITX_DDC_WRITE_blocking(state, &data_in, &data_out);
+		pr_info("CDN_API_HDMITX_DDC_WRITE_blocking ret = %d\n", ret);
+	}
 
 	ret = CDN_API_General_Read_Register_blocking(
 				state, ADDR_SOURCE_MHL_HD + (HDTX_CONTROLLER << 2), &resp);
-- 
1.7.9.5

